Tasks/Common/webdeployment-common/deployusingmsdeploy.ts

```diff
@@ -65,4 +72,25 @@ export async function DeployUsingMSDeploy(webDeployPkg, webAppName, publishingPr
             tl.rmRF(setParametersFile, true);
         }
     }
-}
\ No newline at end of file
+}
+
+export async function executeMSDeploy(msDeployCmdArgs, retryCount, retryInterval, errObj) {
+    var deferred = Q.defer();
+    try {
+        await tl.exec("msdeploy", msDeployCmdArgs, <any>{failOnStdErr: true, errStream: errObj});
+        deferred.resolve("Azure App service successfully deployed");
+    } catch (error) {
```

Comment:
Can we invoke the retry logic, only when it is related to connection terminated.
We can also [inform the user](https://github.com/Microsoft/vsts-tasks/blob/2b77f875ea4ba5f78edd740c3ac28c60c730fcd2/Tasks/Common/webdeployment-common/msdeployutility.ts#L169) that we retry for the above error.